soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -2835,86 +2835,251 @@ void CreateNaviRandoMessages() {
     }
 }
 
-void CreateIceTrapRandoMessages() {
-    CustomMessage IceTrapMessages[NUM_ICE_TRAP_MESSAGES] = {
-        { "You are a %bFOOL%w!", "Du bist ein %bDUMMKOPF%w!", "%bPauvre fou%w..." },
-
-        { "You are a %bFOWL%w!", "Du bist eine %bFrostbeule%w!", "Tu es un %bglaçon%w, Harry!" },
-
-        { "%bFOOL%w!", "%bDUMMKOPF%w!", "%bSot%w que tu es." },
-
-        { "You just got %bPUNKED%w!", "Du wurdest %beiskalt%w erwischt!", "Ça me %bglace%w le sang!" },
-
-        { "Stay %bfrosty%w, @.", "Es läuft Dir %beiskalt%w den Rücken&hinunter, @.", "%bReste au frais%w, @." },
-
-        { "Take a %bchill pill%w, @.", "Bleib %bcool%w, @.", "Et c'est la douche %bfroide%w!" },
-
-        { "%bWinter%w is coming.", "Der %bWinter%w naht.", "L'%bhiver%w vient." },
-
-        { "%bICE%w to see you, @.", "Alles %bcool%w im Pool?", "%bGlacier%w!" },
-
-        { "Feeling a little %rhot%w under the collar?&%bLet's fix that%w.", "%bAbkühlung gefällig%w?",
-          "%Ça en jette un %bfroid%w." },
-
-        { "It's a %bcold day%w in the Evil Realm.", "Es ist ein %kalter%w Tag im Herzen&von Hyrule.",
-          "Est-ce que tu as déjà eu des sueurs&%bfroides%w?" },
-
-        { "Getting %bcold feet%w?", "Bekommst Du etwa %bkalte%w Füße?",
-          "La vengeance est un plat qui se mange&%bfroid%w!" },
-
-        { "Say hello to the %bZoras%w for me!", "Sag den %bZoras%w viele Grüße von mir!",
-          "Dit bonjour aux %bZoras%w pour moi!" },
-
-        { "Can you keep a %bcool head%w?", "Bewahre einen %bkühlen%w! Kopf.",
-          "Il faut parfois savoir garder la tête&%bfroide%w!" },
-
-        { "Ganondorf used %bIce Trap%w!&It's super effective!",
-          "Ganondorf setzt %bEisstrahl%w ein.&Das ist sehr effektiv!",
-          "Ganondorf utilise %bPiège de Glace%w!&C'est super efficace!" },
-
-        { "Allow me to break the %bice%w!", "Ein Lächeln ist der beste Weg,&um das %bEis%w zu brechen!",
-          "Laisse moi briser la %bglace%w!" },
-
-        { "%bCold pun%w.", "%bEiskalt%w lässt du meine Seele&erfrier'n.",
-          "Balance man...,&Cadence man...,&Trace la %bglace%w...,&c'est le Cooooolllll Rasta!" },
-
-        { "The %bTitanic%w would be scared of you,&@.", "Die %bTitanic%w hätte Angst vor Dir,&@.",
-          "Le %bTitanic%w aurait peur de toi,&@." },
-
-        { "Oh no!", "Oh nein!", "Oh non!" },
-
-        { "What killed the dinosaurs?&The %bICE%w age!", "Was die Dinosaurier getötet hat?&Die %bEiszeit%w!",
-          "Qu'est-ce qui a tué les dinosaures?&L'ère %bglacière%w!" },
-
-        { "Knock knock. Who's there? Ice. Ice&who? Ice see that you're a %bFOOL%w.",
-          "Nachts ist es %bkälter%w als draußen.",
-          "L'imbécile réfléchit uniquement quand il&s'observe dans la %bglace%w." },
+CustomMessage Randomizer::GetIceTrapMessage() {
+    static const char* const englishIceTrapMessages[169] = {
```

Comment:
I've never understood why we put static variables inside functions. I know since they're static they shouldn't be instantiated more than once, but it seems like bad design to me.

Reply:
This way they're scoped to the function and can't be accessed outside of it.